tests/Gedmo/Mapping/MappingEventSubscriberTest.php

```diff
@@ -64,10 +68,56 @@ public function testGetMetadataFactoryCacheFromDoctrine(): void
         static::assertTrue($cache->hasItem($cacheKey));
     }
 
+    public function testGetMetadataFactoryCacheFromDoctrineForSuperClassExtension(): void
+    {
+        $metadataFactory = $this->em->getMetadataFactory();
+        $getCache = \Closure::bind(static function (AbstractClassMetadataFactory $metadataFactory): ?CacheItemPoolInterface {
+            return $metadataFactory->getCache();
+        }, null, \get_class($metadataFactory));
+
+        /** @var CacheItemPoolInterface $cache */
+        $cache = $getCache($metadataFactory);
+
+        $cacheKey = ExtensionMetadataFactory::getCacheId(SuperClassExtension::class, 'Gedmo\Tests\Mapping\Mock\Extension\Encoder');
+
+        static::assertFalse($cache->hasItem($cacheKey));
+
+        $subscriber = new EncoderListener();
+        $classMetadata = $this->em->getClassMetadata(SuperClassExtension::class);
+
+        $config = $subscriber->getExtensionMetadataFactory($this->em)->getExtensionMetadata($classMetadata);
+
+        self::assertEquals([
```

Comment:
```suggestion
        self::assertSame([
```
To make the build green